game_patch/object/obj_light.cpp

```diff
@@ -133,6 +134,25 @@ ConsoleCommand2 mesh_static_lighting_cmd{
     "Toggle mesh static lighting calculation",
 };
 
+CallHook<void(rf::Entity*)> entity_update_muzzle_flash_light_hook{
+    0x0041E814,
+    [](rf::Entity* ep) {
```

Comment:
This isn't necessary since the entity is confirmed as valid earlier in the `entity_process_post` function, but I also can't see a downside to passing it as a reference instead, so as a best practice its probably a good idea. Thanks!

Reply:
`entity_update_muzzle_flash_light` takes a reference so a pointer is objectively wrong. But I only know because I checked. The point is that it would lead to a crash if _somebody_ _else_ called it with a null pointer.

Reply:
> `entity_update_muzzle_flash_light` takes a reference so a pointer

Mind if I ask how you checked and confirmed that? I ask because from what I'm seeing in IDA, it seems to take a pointer? Just wondering - if I'm missing something or doing something wrong I'd like to learn for the future 😄 

I'm seeing the void referred to in IDA as:
`void __cdecl entity_update_muzzle_flash_light(Entity *ep)`

Reply:
To be honest I'm not a big fan of `Entity` references. Yes, it makes it clear that it's never null, but it doesn't mean a pointer must be nullable in every context. Considering how RF is coded you deal with `Entity` pointers all the time and constantly switching between pointer and reference is a downside for me.

Reply:
>  Also I rather dislike mutable references, they don't show mutability at the call side and looks like a copy.

Is this flaw not a reality of orthodox C++? An immutable references looks like a copy as well. You might prefer `gsl::not_null<T>` with is authoritative instead of references in this situation. I do not think it makes sense to throw this invariant information away when we have two options.